### ISSUE TITLE
Zoom gesture seems to be "choppy"/steppy

### ISSUE THREAD

**Issue by wareya** [emilk/egui#5384]
Checked with 0.29.1 and master.

**Describe the bug**

When I zoom in on the canvas in my image editing application (https://wareya.github.io/warpainter/) on a phone with a two-finger zoom gesture, the zoom is choppy and jumps in/out every so often, as though it's emulating a scroll wheel at the same time as it's doing the gesture. I'm using the `zoom_delta()` function on `InputState` and then storing the result in my application's own input messaging structure. I also have non-ctrl scroll wheel zoom (this is normal for image editors), but disabled it during testing and could still see the issue, while being unable to use the scroll wheel to zoom in the desktop version. However I could use ctrl+scroll wheel zoom in the desktop version.

https://github.com/user-attachments/assets/98943c94-1ad1-47ce-a7d2-f4c87022b2ba

![image](https://github.com/user-attachments/assets/b6ac6d0e-5eb5-4aee-9be3-f78b20283d79)

tldr: zoom_delta() seems to have both true gesture zoom and also emulated scroll wheel zoom at the same time when doing gesture zoom and there should be a direct way to access them separately from one another.

**To Reproduce**

Go to https://wareya.github.io/warpainter/ on a phone or tablet, do two-finger zoom.

**Expected behavior**
Smooth zoom.

**Screenshots**
See above embedded video.

**Smartphone (please complete the following information):**
 - Device: Pixel 6
 - OS: Android 12
 - Browser both Firefox and Chrome
 - Version 0.29.1 and master
 

EDIT: this issue seems to also apply to the `zoom_delta` property of MultiTouchInfo, minus the ability to zoom on desktop with ctrl+scrollwheel.

**Comment by wareya** [comment]
PEBCAK, I had zoom level rounding code buried elsewhere in my application.
